### ISSUE TITLE
Unable to build Android projects with AOT and LLVM enabled on Windows

### ISSUE THREAD

**Issue by pjcollins** [xamarin/xamarin-android#3120]
### Steps to Reproduce

1. Attempt to install or package an Android project with AOT and LLVM enabled on Windows:
`msbuild /t:SignAndroidPackage *.csproj /p:Configuration=Release /p:AotAssemblies=true /p:EnableLlvm=true`

### Actual Behavior
AOT compilation does not work - [binlog](https://github.com/xamarin/xamarin-android/files/3213948/bad.zip).

```
[AOT] MONO_PATH="C:\Users\peter\source\repos\AppTest\AppTest\obj\Release\90\android\assets" MONO_ENV_OPTIONS="" "C:\Program Files (x86)\Microsoft Visual Studio\2019\Stable\MSBuild\Xamarin\Android\cross-arm" --llvm --aot=msym-dir=C:\Users\peter\source\repos\AppTest\AppTest\obj\Release\90\aot\armeabi-v7a,outfile=C:\Users\peter\source\repos\AppTest\AppTest\obj\Release\90\aot\ARMEAB~1\libaot-Xamarin.Android.Arch.Core.Runtime.dll.so,asmwriter,mtriple=armv7-linux-gnueabi,tool-prefix=C:\PROGRA~2\Android\ANDROI~1\NDK-BU~1\TOOLCH~1\ARM-LI~1.9\prebuilt\WINDOW~1\bin\arm-linux-androideabi-,ld-flags=C:\PROGRA~2\Android\ANDROI~1\NDK-BU~1\TOOLCH~1\ARM-LI~1.9\prebuilt\WINDOW~1\lib\gcc\ARM-LI~1\49696B~1.X\libgcc.a;C:\PROGRA~2\Android\ANDROI~1\NDK-BU~1\PLATFO~1\ANDROI~2\arch-arm\usr\lib\libc.so;C:\PROGRA~2\Android\ANDROI~1\NDK-BU~1\PLATFO~1\ANDROI~2\arch-arm\usr\lib\libm.so,llvm-path=C:\PROGRA~2\MIB055~1\2019\Stable\MSBuild\Xamarin\Android,temp-path=C:\Users\peter\source\repos\AppTest\AppTest\obj\Release\90\aot\ARMEAB~1\Xamarin.Android.Arch.Core.Runtime.dll C:\Users\peter\source\repos\AppTest\AppTest\obj\Release\90\android\assets\Xamarin.Android.Arch.Core.Runtime.dll
```

Using [Dependencies](https://github.com/lucasg/Dependencies), we noticed there is a missing dependency on `libstdc++-6.dll`:

![image](https://user-images.githubusercontent.com/840039/58282888-99b18a00-7d6c-11e9-88ea-4d06d508530d.png)

A working one from VS 2019 16.0 looks like this:

![image](https://user-images.githubusercontent.com/840039/58282955-c2398400-7d6c-11e9-91e9-4ae2d7d4561b.png)

### Version Information
d16-2 - 9.4.0.2

**Comment by lambdageek** [comment]
We have a PR that should address this https://github.com/mono/mono/pull/14283.  I need to update it based on feedback and backport to 2019-02.

**Comment by jonpryor** [comment]
Probable (in-progress) fix: https://github.com/mono/mono/pull/14283

**Comment by lambdageek** [comment]
Should be fixed once XA picks up https://github.com/mono/mono/pull/14668

**Comment by brendanzagaeski** [comment]
For searchability, I'll add that the final error that appears at the end of the attached `.binlog` file is:

```
error XA3001: Could not AOT the assembly: Xamarin.Android.Arch.Lifecycle.LiveData.dll
```

(It took me a bit time to find this report because the "could not aot the assembly" was not yet mentioned here.)

**Comment by tranb3r** [comment]
With the last VS update (16.2.0 preview 2.0), I cannot build anymore with AOT enabled.
Is this issue fixed or not ? (the ticket is closed)
How come a new VS preview has been released, knowing that it would break xamarin.android build ??

**Comment by brendanzagaeski** [comment]
Thanks much for trying the preview! Apologies in advance that I don't have details on the constraints of the release process, but to add some information I have:

The fix for this issue is indeed *not* yet included in the Xamarin.Android SDK version 9.4.0.17 that is included in Visual Studio 2019 version 16.2 Preview 2. But the fix is included in the current continuous integration builds targeting the upcoming previews, so barring any unexpected complications, it *will* be included in the next version of the Xamarin.Android SDK in Visual Studio 2019 version 16.2 Preview 3.

In case it be might be useful for reference, this item is included as a [known issue](https://docs.microsoft.com/xamarin/android/release-notes/9/9.4#known-issues-1) in the release notes for Xamarin.Android 9.4.0.17, and it will be included as an "issue fixed" in the release notes for the next released Xamarin.Android SDK preview version (that will include the fix).

(When I was away from the computer yesterday after adding my previous comment, I also had a thought that I should come back to this issue and add a mention of the 9.4.0.17 version in particular to provide additional visibility on the known temporary breakage in that version, but unfortunately I forgot that idea by the time I got back to the computer. Thanks for the reminder!)

**Comment by tranb3r** [comment]
Thanks @brendanzagaeski for the clarification.
Too bad this was not mentionned in the Visual Studio 16.2 preview 2 release notes... 
Next time, I think it'd be worth mentionning _at least_ the fact that Xamarin.Android version has changed (and also add a link to the corresponding release notes).
Do you know when preview 3 will be released ?

**Comment by brendanzagaeski** [comment]
I'm not too sure of the timeline for Visual Studio 2019 version 16.2 Preview 3, but so far the time between the various Visual Studio 2019 version 16.1 and version 16.2 Preview versions updates has been [at most 3 weeks](https://docs.microsoft.com/visualstudio/install/visual-studio-build-numbers-and-release-dates?view=vs-2019), so I suspect the time until the next preview will fall within that range too.

**Comment by tranb3r** [comment]
Visual Studio 2019 version 16.2 Preview 3 is out.
Is the android aot build fixed in this release? (and what is the corresponding Xamarin.Android SDK version ?)

**Comment by brendanzagaeski** [comment]
Yes, this fix has now been published.

**Windows fix published**. The new Xamarin.Android SDK version 9.4.0.34 preview that includes a fix for this issue has now been published as part of Visual Studio 2019 version 16.2 Preview 3. [Check for the latest updates](https://docs.microsoft.com/visualstudio/install/update-visual-studio?view=vs-2019) or install the most recent Preview release from <https://visualstudio.microsoft.com/downloads/> to get the fix.

**macOS fix published**. The new Xamarin.Android SDK version 9.4.0.34 preview that includes a fix for this issue has now been released in the Preview updater channel in Visual Studio for Mac. [Check for the latest updates](https://docs.microsoft.com/visualstudio/mac/update) or install the most recent release from <https://visualstudio.microsoft.com/downloads/> and then [check for the latest updates](https://docs.microsoft.com/visualstudio/mac/update) on the Preview channel to get the fix.
